flight/Modules/Buzzer/buzzer.c

```diff
@@ -0,0 +1,260 @@
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ *
+ * Additional note on redistribution: The copyright and license notices above
+ * must be maintained in each individual source file that is a derivative work
+ * of this source file; otherwise redistribution is prohibited.
+ */
+
+#include "openpilot.h"
+
+#include "flightbuzzersettings.h"
+#include "modulesettings.h"
+#include "flightstatus.h"
+#include "accessorydesired.h"
+#include "pios_thread.h"
+
+// ****************
+// Private constants
+#define STACK_SIZE_BYTES            420  //gcs shows ~40 stack remaining
+#define TASK_PRIORITY               PIOS_THREAD_PRIO_LOW
+#define SAMPLE_PERIOD_MS            200
+// Private types
+
+// Private variables
+static bool module_enabled = false;
+static struct pios_thread *buzzerTaskHandle;
+static int8_t buzzerPin = -1; //Pin for buzzer
+static int8_t accChan = -1; //Accessory Channel
+static bool buzzer_settings_updated;
+static bool wasArmed;
+static bool wasDisarmed;
+static 	FlightBuzzerSettingsData buzzerSettings;
+
+// ****************
+// Private functions
+static void buzzerTask(void * parameters);
+static void getTune(uint8_t arming, uint32_t *newTune);
+
+static int32_t BuzzerStart(void)
+{
+	if (module_enabled) {
+		//FlightBuzzerSettingsConnectCallbackCtx(UAVObjCbSetFlag, &buzzer_settings_updated);
+
+		// Start tasks
+		buzzerTaskHandle = PIOS_Thread_Create(buzzerTask, "buzzerBridge", STACK_SIZE_BYTES, NULL, TASK_PRIORITY);
+		TaskMonitorAdd(TASKINFO_RUNNING_BUZZER, buzzerTaskHandle);
+		return 0;
+	}
+	return -1;
+}
+/**
+ * Initialise the module, called on startup
+ * \returns 0 on success or -1 if initialisation failed
+ */
+int32_t BuzzerInitialize(void)
+{
+#ifdef MODULE_Buzzer_BUILTIN
+	module_enabled = true;
+#else
+	uint8_t module_state[MODULESETTINGS_ADMINSTATE_NUMELEM];
+	ModuleSettingsAdminStateGet(module_state);
+	if (module_state[MODULESETTINGS_ADMINSTATE_BUZZER] == MODULESETTINGS_ADMINSTATE_ENABLED) {
+		module_enabled = true;
+	} else {
+		module_enabled = false;
+		return 0;
+	}
+#endif
+	FlightBuzzerSettingsInitialize();
+
+	return 0;
+}
+MODULE_INITCALL(BuzzerInitialize, BuzzerStart);
+
+/**
+ * Main task. It does not return.
+ */
+static void buzzerTask(void * parameters)
+{
+	
+	buzzer_settings_updated = true;
+	wasArmed = false;
+	wasDisarmed = true;
+
+	// Main task loop
+	uint32_t lastSysTime;
+	lastSysTime = PIOS_Thread_Systime();
+	
+	while (true) {
+		PIOS_Thread_Sleep_Until(&lastSysTime, SAMPLE_PERIOD_MS);
+
+		if (buzzer_settings_updated) {
+			buzzer_settings_updated = false;
+			FlightBuzzerSettingsGet(&buzzerSettings);
+
+			accChan = buzzerSettings.AcessoryChan;
```

Comment:
No need to use separate variables for these, just use `buzzerSettings.AcessoryChan` etc directly in the code